core/src/main/java/io/grpc/Metadata.java

```diff
@@ -362,17 +364,30 @@ private String toStringInternal() {
    *
    * <p>Only the following ASCII characters are allowed in the names of keys:
    * <ul>
-   *   <li>letters, i.e., {@code a-z} and {@code A-Z}</li>
-   *   <li>digits, i.e., {@code 0-9}</li>
-   *   <li>dash, i.e., {@code -}</li>
-   *   <li>underscore, i.e., {@code _}</li>
+   *   <li>letters: {@code a-z}</li>
```

Comment:
> Comparing characters in a case insensitive manner is one thing, but changing their values for them is another

Why is that? The user never sees the string we turn it into. They only ever use a Key.